src/coreclr/minipal/Unix/CMakeLists.txt

```diff
@@ -6,6 +6,7 @@ set(SOURCES
 if(NOT CLR_CROSS_COMPONENTS_BUILD)
     list(APPEND SOURCES
         ${CLR_SRC_NATIVE_DIR}/minipal/cpufeatures.c
+        ${CLR_SRC_NATIVE_DIR}/minipal/utf8.c
     )
 endif()
```

Comment:
```diff
if(NOT CLR_CROSS_COMPONENTS_BUILD)
    list(APPEND SOURCES
        ${CLR_SRC_NATIVE_DIR}/minipal/cpufeatures.c
-        ${CLR_SRC_NATIVE_DIR}/minipal/utf8.c
    )
+
+    if(CLR_CMAKE_HOST_OSX)
+        list(APPEND SOURCES
+            ${CLR_SRC_NATIVE_DIR}/minipal/utf8.c
+        )
+    endif()
endif()
```

Reply:
Special casing osx looks too weird. I'd like to  see if there's a better solution.

Reply:
yea, mscordbi does weird things to make pal, dacpal, minipal paths work e.g. https://github.com/dotnet/runtime/blob/94f411c6ce7354a49525d0cd2cadefffcbc66f37/src/coreclr/dlls/mscordbi/CMakeLists.txt#L121-L124 so no surprise if we need another special case to make this non-standard stuff work satisfying all linkers alike.

Reply:
@AaronRobinsonMSFT any input on this?

Reply:
What is the failure that is occurring?

Reply:
Linux object file has the utf8 symbols exposed from coreclr-pal so mscordbi project finds it in second round of linking (see the linked hack), while OSX object file doesn't expose utf8 symbols and linking fails. As a result, if we don't make changes in this file, we get missing symbols error on OSX and no error on linux. With the current change, OSX linker (ld64) is happy but linux linker give us "duplicate symbols" error. With the proposed change, both linkers are happy.

AFAIK, the long-term plan is to use standard library in coreclr and slim down the PAL. Until that happens we will need to rely on these hacks. (I have shown above the other hack we are using in mscordbi).

Reply:
Thanks @am11. The duplicate symbol issue is sadly annoying. I agree with your recommendation here on how to avoid it.

> AFAIK, the long-term plan is to use standard library in coreclr and slim down the PAL.

Yep. I am hoping to take a big step in that direction this release. You will be tagged :-)